src/ServiceControl/Monitoring/InMemoryMonitoringDataStore.cs

```diff
@@ -71,6 +71,28 @@ public Task Delete(Guid endpointId)
             return Task.CompletedTask;
         }
 
+        public Task BulkCreate(EndpointDetails[] newEndpoints)
+        {
+            foreach (var endpoint in newEndpoints)
+            {
+                var id = DeterministicGuid.MakeId(endpoint.Name, endpoint.HostId.ToString());
```

Comment:
We seem to be doing this all the time. Why not have this encoded as a proper ID in the endpoint details? then we can do it once and use it across the board